### PR TITLE
#2393 BugSplat Crash #1504586: LLRenderTarget::getTexture(483)

### PR DESCRIPTION
**Pull request by LLGuru**

### CODE REVIEW

File: indra/llrender/llrendertarget.cpp [hunk 1]
```diff
@@ -475,7 +475,7 @@ void LLRenderTarget::clear(U32 mask_in)
 
 U32 LLRenderTarget::getTexture(U32 attachment) const
 {
-    if (attachment > mTex.size()-1)
```
Comment (akleshchev) [inline indra/llrender/llrendertarget.cpp:478]: What's the point of this change?

Comment (LLGuru) [reply indra/llrender/llrendertarget.cpp:478]: > What's the point of this change?

It doesn't work in case of an empty container

Comment (akleshchev) [reply indra/llrender/llrendertarget.cpp:478]: Right, unsigned int under the hood, so zero minus one won't work correctly and looks like it can be empty.

**Review by akleshchev — APPROVED** [review]

File: indra/llrender/llrendertarget.cpp (hunk 1 above)
Comment (akleshchev) [reply indra/llrender/llrendertarget.cpp:478]: But why does it crash?
Even if condition doesn't work, execution would check mTex.empty() and return zero, not crash on getTexture().

Comment (LLGuru) [reply indra/llrender/llrendertarget.cpp:478]: Not sure how the optimizer transforms such ugly code

**Review by RunitaiLinden — APPROVED** [review]

**Pull request merged**